### PR TITLE
updated gradles, dependencies, and some code

### PR DESCRIPTION
**Pull request by TacoTheDank**
Signed-off-by: TacoTheDank <SkytkRSfan3895@gmail.com>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,6 @@
...
-.gradle
-/local.properties
-/.idea
-.DS_Store
-/build
-/captures
+.gradle/
+local.properties
+.idea/
+app/build/
```
Comment (fython) [inline .gitignore]: Why removing `.gitignore` in `app` module and adding `app/build/` here? With the old style, we needn't modify anything in this `.gitignore` if I change the module name.

Comment (TacoTheDank) [reply .gitignore]: hmm alright (but then what kind of thing would require changing the module name?)

**Comment by TacoTheDank** [comment]
thats fine

**Comment by TacoTheDank** [comment]
🤔

**Pull request closed without merging**